pkg/controller/statusmanager/status_manager.go

```diff
@@ -85,6 +85,12 @@ func (status *StatusManager) deleteRelatedObjectsNotRendered(co *configv1.Cluste
 				status.relatedObjects = append(status.relatedObjects, currentObj)
 				continue
 			}
+			if gvk.Kind == "Namespace" {
```

Comment:
Check that gvk.Group == "core" too.

Reply:
In the namespace `gvk` object, the group is empty string, so I change use `gvk.Group == ""` here.